katsdpcontroller/schemas/s3_config.json

```diff
@@ -1,65 +1,80 @@
 {
-    "$schema": "http://json-schema.org/draft-04/schema#",
-    "definitions": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$defs": {
         "key": {"type": "string"},
-        "key_pair": {
+        "has_keys": { "required": ["access_key", "secret_key"] },
+        "has_url": { "required": ["url"] },
+        "no_url": { "not": { "required": ["url"] } },
+        "section": {
+            "$comment": "A read or write section in a store (no required properties)",
             "type": "object",
             "additionalProperties": false,
             "properties": {
-                "access_key": {"$ref": "#/definitions/key"},
-                "secret_key": {"$ref": "#/definitions/key"}
+                "access_key": {"$ref": "#/$defs/key"},
+                "secret_key": {"$ref": "#/$defs/key"},
+                "url": { "type": "string", "format": "uri" }
+            },
+            "if": {
```

Comment:
Is there a use key for URL with no keys ?

Reply:
Yes e.g. see the sandbox s3_config.json: we don't directly interact with the archive (yet) so we don't need any keys for it; but we need to know the read URL so that it can be embedded in telstate.